src/v/kafka/server/handlers/fetch.cc

```diff
@@ -177,13 +177,14 @@ static ss::future<read_result> read_from_partition(
  */
 static ss::future<read_result> do_read_from_ntp(
   cluster::partition_manager& mgr,
+  cluster::metadata_cache& md_cache,
   ntp_fetch_config ntp_config,
   bool foreign_read,
   std::optional<model::timeout_clock::time_point> deadline) {
     /*
      * lookup the ntp's partition
      */
-    auto partition = mgr.get(ntp_config.ntp());
+    auto partition = make_partition_proxy(ntp_config.ntp(), md_cache, mgr);
```

Comment:
maybe we can stick to `kafka_partition` name since proxy is kafka layer abstraction ?